drivers/video/video.c

```diff
@@ -972,16 +993,20 @@ static void cleanup_scenes_parameter(void)
 
 static void cleanup_resources(FAR video_mng_t *vmng)
 {
-  /* clean up resource */
+  /* If in capture, stop */
 
-  if ((vmng->video_inf.state == VIDEO_STATE_CAPTURE) ||
-      (vmng->still_inf.state == VIDEO_STATE_CAPTURE))
+  if (vmng->video_inf.state == VIDEO_STATE_CAPTURE)
```

Comment:
There are logic to avoid both stream enter capture state, please reference here:
https://github.com/apache/nuttx/pull/7787/files#diff-1a26ae0c754d5e6afb90f3e5e58a10cd3054e4ae37ab148284b8b6513baa8af4R452-R454